middlewares/osio/auth.go

```diff
@@ -246,3 +247,38 @@ func cacheKey(plainKey string) string {
 	hash := hex.EncodeToString(h.Sum(nil))
 	return hash
 }
+
+func extractUserID(req *http.Request) string {
+	userID := ""
+	if req.Header.Get(UserIDHeader) != "" {
+		userID = req.Header.Get(UserIDHeader)
+	} else if req.URL.Query().Get(UserIDParam) != "" {
+		userID = req.URL.Query().Get(UserIDParam)
+		if strings.Contains(userID, "/") {
```

Comment:
@nurali-techie this code seems to be unreachable for urls like `https://oso-master-url/?param=value/api/v1/namespaces/namespace/serviceaccounts`

Reply:
go consider such url having `param` as query parameter with value `value/api/v1/namespaces/namespace/serviceaccounts` so above url will work.  Test case added please check [here](https://github.com/fabric8-services/fabric8-oso-proxy/pull/23/files#diff-6de73e373942d5306d5a8beef28a9e54R56)

Reply:
ah, good to know